Feeder audit 2026-05-29: snotel

## Feeder audit summary for `snotel`

This sweep covers the objective repo-state parts of the feeder-release-checklist: required file presence, catalog + ghpages source entries, source-local pytest with generated producer src paths injected into PYTHONPATH, Docker E2E matrix coverage, ARM static audit, Fabric static audit, and ARM generator dry-run no-diff. It does not rerun live Azure/Fabric deploy bullets, specialist-agent reviews, or upstream API re-audits.

## Findings

- **ERROR** `missing-docker-build-matrix-entry` — 1 finding(s)
  - tests/docker_e2e/matrix.json has no build entry for kafka (Dockerfile). (tests/docker_e2e/matrix.json)
- **ERROR** `missing-docker-flow-matrix-entry` — 1 finding(s)
  - tests/docker_e2e/matrix.json has no flow entry for kafka (Dockerfile). (tests/docker_e2e/matrix.json)
- **ERROR** `pytest-failed` — 1 finding(s)
  - Source-local pytest failed. (feeders/snotel/tests)
- **WARNING** `arm-stub-description` — 2 finding(s)
  - azure-template-with-eventhub.json: parameter 'eventHubSku' has stub or missing metadata.description (feeders/snotel/azure-template-with-eventhub.json)
  - azure-template-with-servicebus.json: parameter 'queueName' has stub or missing metadata.description (feeders/snotel/azure-template-with-servicebus.json)

## Check evidence

### `arm_audit` — rc=0, 1.37s

```text
ARM template static audit
  RepoRoot:        C:\git\real-time-sources
  Feeders scanned: 1
  Templates:       5
  Blockers:        0
  Warnings:        2

[snotel] blockers=0 warnings=2
  azure-template-with-eventhub.json
    [WARNING] stub-description: parameter 'eventHubSku' has stub or missing metadata.description
  azure-template-with-servicebus.json
    [WARNING] stub-description: parameter 'queueName' has stub or missing metadata.description
PASS (with 2 warning(s)).
Report written to C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\arm\snotel.json
```

### `arm_generator` — rc=0, 3.18s

```text
Description sources: total=177, container=31, readme=0, argparse=28, fallback=118
Generated 0 ARM template(s).
```

### `fabric_audit` — rc=0, 1.81s

```text
[32;1mSource[0m[32;1m OptIn[0m[32;1m Notebook[0m[32;1m  Hook[0m[32;1m PASS[0m[32;1m WARN[0m[32;1m BLOCK[0m
[32;1m------[0m [32;1m-----[0m [32;1m--------[0m [32;1m ----[0m [32;1m----[0m [32;1m----[0m [32;1m-----[0m
snotel  True     True False   10    0     0

Sources: 1 total, 0 blockers, 0 warnings
JSON report: C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\fabric\snotel.json
```

### `pytest` — rc=1, 4.88s

```text
============================= test session starts =============================
platform win32 -- Python 3.11.9, pytest-8.4.2, pluggy-1.6.0
rootdir: C:\git\real-time-sources\feeders\snotel
configfile: pytest.ini
plugins: anyio-4.9.0, asyncio-1.2.0, cov-6.0.0, timeout-2.4.0, requests-mock-1.12.1
asyncio: mode=Mode.STRICT, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 64 items

tests\test_once_mode.py ...                                              [  4%]
tests\test_snotel_unit.py ......................F

================================== FAILURES ===================================
________________ TestParseObservationRow.test_parse_valid_row _________________
tests\test_snotel_unit.py:181: in test_parse_valid_row
    obs = parse_observation_row("838:CO:SNTL", row)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
snotel\snotel.py:212: in parse_observation_row
    return SnowObservation(
C:\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\dataclasses_json\undefined.py:117: in _ignore_init
    original_init(self, **final_parameters)
E   TypeError: SnowObservation.__init__() missing 1 required keyword-only argument: 'state'
=========================== short test summary info ===========================
FAILED tests/test_snotel_unit.py::TestParseObservationRow::test_parse_valid_row
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
======================== 1 failed, 25 passed in 1.09s =========================
```

## Next step

Fix the findings above, then rerun the sequential feeder-release-checklist audit for this feeder and close this issue with the new evidence.

--- a/feeders/snotel/snotel/snotel.py
+++ b/feeders/snotel/snotel/snotel.py
@@ -215,9 +215,14 @@ def parse_observation_row(station_triplet: str, row: List[str]) -> Optional[Snow
     prec = parse_float(row[2]) if len(row) > 2 else None
     tobs = parse_float(row[3]) if len(row) > 3 else None
     snwd = parse_float(row[4]) if len(row) > 4 else None
+    state = STATION_METADATA.get(station_triplet, {}).get("state")
+    if state is None:
+        parts = station_triplet.split(":")
+        state = parts[1] if len(parts) > 1 else ""
 
     return SnowObservation(
         station_triplet=station_triplet,
+        state=state,
         date_time=dt,
         snow_water_equivalent=swe,
         snow_depth=snwd,

--- a/feeders/snotel/tests/test_snotel_unit.py
+++ b/feeders/snotel/tests/test_snotel_unit.py
@@ -181,6 +181,7 @@ class TestParseObservationRow:
         obs = parse_observation_row("838:CO:SNTL", row)
         assert obs is not None
         assert obs.station_triplet == "838:CO:SNTL"
+        assert obs.state == "CO"
         assert obs.snow_water_equivalent == 9.0
         assert obs.precipitation == 16.40
         assert obs.air_temperature == 36.3
@@ -579,6 +580,7 @@ class TestDataClasses:
         dt = datetime(2026, 4, 8, 0, 0, tzinfo=timezone.utc)
         obs = SnowObservation(
             station_triplet="838:CO:SNTL",
+            state="CO",
             date_time=dt,
             snow_water_equivalent=9.0,
             snow_depth=26.0,
@@ -586,6 +588,7 @@ class TestDataClasses:
             air_temperature=36.3,
         )
         assert obs.station_triplet == "838:CO:SNTL"
+        assert obs.state == "CO"
         assert obs.snow_water_equivalent == 9.0
         assert obs.date_time == dt
 
@@ -593,6 +596,7 @@ class TestDataClasses:
         dt = datetime(2026, 4, 8, 0, 0, tzinfo=timezone.utc)
         obs = SnowObservation(
             station_triplet="838:CO:SNTL",
+            state="CO",
             date_time=dt,
             snow_water_equivalent=None,
             snow_depth=None,
@@ -608,6 +612,7 @@ class TestDataClasses:
         dt = datetime(2026, 4, 8, 0, 0, tzinfo=timezone.utc)
         obs = SnowObservation(
             station_triplet="838:CO:SNTL",
+            state="CO",
             date_time=dt,
             snow_water_equivalent=9.0,
             snow_depth=26.0,
@@ -622,6 +627,7 @@ class TestDataClasses:
         dt = datetime(2026, 4, 8, 0, 0, tzinfo=timezone.utc)
         obs = SnowObservation(
             station_triplet="838:CO:SNTL",
+            state="CO",
             date_time=dt,
             snow_water_equivalent=9.0,
             snow_depth=26.0,

